src/termdict/merger.rs

```diff
@@ -114,3 +114,64 @@ impl<'a> TermMerger<'a> {
         &self.current_streamers[..]
     }
 }
+
+#[cfg(all(test, feature = "unstable"))]
+mod bench {
+    use super::TermMerger;
+    use crate::directory::FileSlice;
+    use crate::postings::TermInfo;
+    use crate::termdict::{TermDictionary, TermDictionaryBuilder};
+    use rand::distributions::Alphanumeric;
+    use rand::{thread_rng, Rng};
+    use test::{self, Bencher};
+
+    fn make_term_info(term_ord: u64) -> TermInfo {
+        let offset = |term_ord: u64| (term_ord * 100 + term_ord * term_ord) as usize;
+        TermInfo {
+            doc_freq: term_ord as u32,
+            postings_range: offset(term_ord)..offset(term_ord + 1),
+            positions_idx: offset(term_ord) as u64 * 2u64,
+        }
+    }
+
+    /// Create a dictionary of random strings.
+    fn rand_dict(size: usize) -> crate::Result<TermDictionary> {
+        let buffer: Vec<u8> = {
+            let mut terms = vec![];
+            for _i in 0..size {
+                let rand_string: String = thread_rng()
+                    .sample_iter(&Alphanumeric)
+                    .take(30)
```

Comment:
Can we have a variable number of chars?

Due to branch prediction it might strongly affect the performance.